pyvista/plotting/plotting.py

```diff
@@ -4762,36 +4797,50 @@ def add_point_labels(
             points = np.array(points)
 
         if isinstance(points, np.ndarray):
-            vtkpoints = pyvista.PolyData(points)  # Cast to poly data
-        elif is_pyvista_dataset(points):
-            vtkpoints = pyvista.PolyData(points.points)
-            if isinstance(labels, str):
-                labels = points.point_data[labels]
-        else:
+            points = pyvista.PolyData(points)  # Cast to poly data
+        elif not is_pyvista_dataset(points) and not isinstance(points, _vtk.vtkAlgorithm):
```

Comment:
I think the docstring also needs to be amended with algorithms.

Reply:
Done in 3feddccec87b7420e12b2ab3ddb224852f6c84eb